<launch>  

  <!-- upload some parameters on the parameter server -->
  <group ns="towr">
    <param name="rosbag_folder" value="$(find towr_ros)/bags" type="string"/>   
  </group> 

  <!-- translate keyboard input to TOWR commands -->
  <node 
    name="user_interface" 
    pkg="towr_ros" 
    type="towr_user_interface" 
    output="screen"
    launch-prefix="xterm -e">  <!-- open in new window -->
  </node>
  
  <!-- draw the different terrains in RVIZ -->
  <node 
    name="rviz_terrain_publisher" 
    pkg="towr_ros" 
    type="rviz_terrain_publisher" 
    output="screen">
  </node>
  
  <!--  visualize the goal pose in RVIZ -->
  <node 
    name="goal_pose_publisher" 
    pkg="towr_ros" 
    type="goal_pose_publisher" 
    output="screen">
  </node>
    
  <!-- start the actual NLP solver using TOWR -->
  <node 
    name="towr_ros" 
    pkg="towr_ros" 
    type="towr_ros" 
    output="screen">
  </node>
    
</launch>
